lib/kubeclient/common.rb

```diff
@@ -286,6 +308,45 @@ def get_entities(entity_type, klass, resource_name, options = {})
       Kubeclient::Common::EntityList.new(entity_type, resource_version, collection)
     end
 
+    def get_entities_async(klass, resource_name, options = {})
+      params = {}
+      SEARCH_ARGUMENTS.each { |k, v| params[k] = options[v] if options[v] }
+
+      ns_prefix = build_namespace_prefix(options[:namespace])
+
+      resource_version = nil
+      entity_streamer = Json::Streamer::JsonStreamer.new
+      resource_version_streamer = Json::Streamer::JsonStreamer.new
+
+      resource_version_streamer.get(key: 'resourceVersion') do |value|
+        resource_version = value unless resource_version
+      end
+
+      # result['items'] might be nil due to https://github.com/kubernetes/kubernetes/issues/13096
+      entity_streamer.get(nesting_level: 2, yield_values: false) do |object|
```

Comment:
What makes this handle "items" specifically?  is it just the level 2?  (if so why doesn't that also get "metadata" values?)
In principle, one day kubernetes might return other nested fields, I'd prefer to ask for "items" specifically.

Reply:
unfortunately, we can't just ask for items specifically: https://github.com/thisismydesign/json-streamer/issues/7#issuecomment-329816147

`nesting_level:2` is the only way to return object-by-object using json-streamer. it may be prudent to write our own implementation at this point (to prevent such a regression)

Reply:
I think rather this one is the applicable comment: https://github.com/thisismydesign/json-streamer/issues/7#issuecomment-330229144